### PR TITLE
Move EAM to Interpolations.jl

### PR DESCRIPTION
**Pull request by cortner**
Hi James  /  @jameskermode

I think this is almost ready, but I got stuck importing an FS models where the knots aren't equi-spaced. Too tired now to pursue this ... Options: 
* switch to a gridded interpolation when the knots aren't equi-spaced
* check how far from equispaced they are and do some surgery 

@jamesgardner1421  - we are dropping Dierckx for Interpolations since the performance of Dierckx with full accuracy is horrible. It would be useful to get your input as well if you are able to comment.

Christoph

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -12,11 +12,13 @@ DelimitedFiles = "8bb1440f-4735-579b-a4ab-409b98df4dab"
 Dierckx = "39dd38d3-220a-591b-8e3c-4c3a8c710a94"
```
Comment (jameskermode) [inline Project.toml:12]: I guess this should be removed?

Comment (cortner) [reply Project.toml:12]: yes, once everything runs ok!

**Comment by jameskermode** [comment]
Looks good from my initial tests wrt LAMMPS, although my test potential doesn't actually have degenerate knots so I'll check with the one you use in the CI tests too.

```julia
using LinearAlgebra
using PyCall
using JuLIP

@pyimport ase.units as ase_units
@pyimport ase.calculators.lammpslib as lammpslib

eam1 = EAM("W.eam.fs")

eam2 = ASECalculator(lammpslib.LAMMPSlib(lmpcmds=["pair_style eam/fs",
                                         "pair_coeff * * W.eam.fs W"],
                                         atom_types=Dict("W" => 1), keep_alive=true))

unitcell = bulk(:W, cubic=true)

@show energy(eam1, unitcell)
@show energy(eam2, unitcell)
@assert energy(eam1, unitcell) ≈ energy(eam2, unitcell)
```

passes, although that's similar to what you already have. 

Lattice constant also fine:

```julia
for (label, eam) in zip([:JuLIP, :LAMMPS], [eam1, eam2])

    unitcell = bulk(:W, cubic=true)
    variablecell!(unitcell)
    set_calculator!(unitcell, eam)
    
    res = optimize(x -> energy(unitcell, x), x -> gradient(unitcell, x), 
                   dofs(unitcell), inplace=false)
    
    alat = unitcell.cell[1, 1]
    @show label, alat
end
```

gives

```
(label, alat) = (:JuLIP, 3.165169502200612)
(label, alat) = (:LAMMPS, 3.16516950220059)
```

Elastic moduli error is dominated by finite differences gradients, so also fine:

```julia
julia> voigt_moduli(eam1, unitcell) ./ ase_units.GPa
6×6 Matrix{Float64}:
 514.296     211.208     211.198      -0.201915   -0.201867   -0.345664
 211.208     514.277     211.188      -0.210202   -0.353151   -0.201802
 211.198     211.188     514.257      -0.360725   -0.210128   -0.210117
  -0.174968   -0.233573   -0.357476  164.681      -0.286978   -0.317033
  -0.174928   -0.349894   -0.233498   -0.283644  164.677      -0.32654
  -0.342403   -0.174855   -0.233467   -0.29214    -0.329869  164.666

julia> voigt_moduli(eam2, unitcell) ./ ase_units.GPa
6×6 Matrix{Float64}:
 514.362     211.153     211.143      -0.218262   -0.218214   -0.373689
 211.153     514.342     211.132      -0.226562   -0.381218   -0.218148
 211.143     211.132     514.323      -0.38876    -0.226455   -0.226427
  -0.189316   -0.251891   -0.385488  164.647      -0.310591   -0.342597
  -0.189313   -0.377983   -0.25182    -0.307296  164.643      -0.352113
  -0.370426   -0.189219   -0.251754   -0.315742   -0.355422  164.632

julia> (voigt_moduli(eam1, unitcell) .- voigt_moduli(eam2, unitcell)) ./ ase_units.GPa
6×6 Matrix{Float64}:
 -0.0656801   0.0557123   0.0556826  0.0163394  0.0163333  0.0280391
  0.0557123  -0.0657634   0.0556292  0.0162992  0.0280124  0.016312
  0.0556826   0.0556292  -0.0658348  0.0279858  0.0162805  0.0162892
  0.0143729   0.0182869   0.0280004  0.0334782  0.0235684  0.0255413
  0.0143358   0.0279977   0.0182446  0.0235623  0.0334728  0.0255368
  0.0280569   0.0143542   0.0182856  0.0235676  0.0255438  0.0334582
  ```

I can add some of these as additional tests if you like?

**Comment by jameskermode** [comment]
Surface energy also fine, so I think we're all good for this potential. Will check with one or two others.

**Comment by cortner** [comment]
that would be really nice to add these tests, thank you! The comparison numbers need to be hard-coded though.

so the branch is usable already and it's just a matter of making it robust and user-friendly before tagging?

**Comment by jameskermode** [comment]
> so the branch is usable already and it's just a matter of making it robust and user-friendly before tagging?

yes, I think so, but still want to test a bit more.

**Comment by jameskermode** [comment]
That commit adds tests of energy, lattice constant, and surface energy. I left out the elastic moduli as that requires JuLIPMaterials (and depends on some compatibility changes and bug-fixes that I still need to make a JuLIPMaterials PR for)

**Comment by jameskermode** [comment]
I removed loading of a non-equispaced `.plt` file in `run_tests.jl`, which didn't seem to be used anywhere. All the `.fs` files I have to hand are already equispaced. So from my point of view I think this is fine to be merged - thanks very much!

**Comment by jameskermode** [comment]
somewhat related, tests don't seem to be running in CI anymore, I guess since Travis went non-free. I could switch to GitHub actions if you like? (in a separate PR)

**Comment by cortner** [comment]
that would be really helpful, thank you.

**Comment by cortner** [comment]
so the plan would be to just enforce equi-spaced nodes for now, and maybe add gridded splines as a backup in the future if somebody requests it?

**Comment by jameskermode** [comment]
Yes, that will work fine. I think non-equispaced files must be somewhat rare

**Comment by jameskermode** [comment]
Oops, accidentally pushed my temporary disabling of the non-EAM tests. Fix coming.

**Comment by cortner** [comment]
ok, so I could now fix the hessian problems on this branch, yes?

**Comment by cortner** [comment]
I suspect that I just have to remove the hack I needed to Dierckx, and it will "just work". We will see.

**Comment by jameskermode** [comment]
I'm hoping it will now pass, let's see

**Comment by cortner** [comment]
I think we have somewhat different behaving versions? There is definitely a test that is using `eam_W` : 
```
at11 = set_pbc!( bulk(:W, cubic = true), false ) * 2
push!(calculators, (eam_W, at11))
```
I've now removed it

**Comment by cortner** [comment]
hm ... I think we are working in parallel :) . Should I stop?

**Comment by jameskermode** [comment]
> I think we have somewhat different behaving versions? 

No, I just didn't check properly - I assumed all the EAM tests were in `test_eam.jl`. I also just pushed the removal of that test.

**Comment by jameskermode** [comment]
I'm done for tonight now, so I'll let you take over. The nightly failure is something different and probably we can just remove the nightly tests from the CI.

**Comment by jameskermode** [comment]
OK, CI passing on 1.6 so I'll call it a night. I'll let you decide what to do about the nightly failures, seems to be some kind of change in behaviour of BoundsError that would bite us eventually so could be worth fixing now.

**Comment by cortner** [comment]
I don't really want to deal with the nightly build failure - it seems it has something to do with Interpolations.jl how the splines are managed?

**Comment by cortner** [comment]
maybe we still need to put version bounds for Interpolations but otherwise it seems good to go from my end.

**Comment by jameskermode** [comment]
I've added a version bound for Interpolations and removed the nightly checks. CI now passing so this is ready to merge, after which #141 can simply be deleted.

This PR closes #139.

**Pull request merged**